pkg/agent/sysadvisor/plugin/qosaware/resource/cpu/advisor.go

```diff
@@ -520,3 +533,31 @@ func (cra *cpuResourceAdvisor) setPoolRegions(poolName string, regions []region.
 	}
 	return cra.metaCache.SetPoolInfo(poolName, pool)
 }
+
+// shareRegionRequirement refers to the resource demand of each share region,
+// sharePoolSize represents the total capacity of the share pool.
+// If the total resource demand of all share regions exceeds the total capacity of the pool,
+// we need to distribute the total capacity proportionally according to the resource requirement.
+func genShareRegionPools(shareRegionRequirement map[string]int, sharePoolSize int) map[string]int {
```

Comment:
It seems that share pool sizes are changed in any case, while we only want to do that when the sum of share pool sizes exceeds their maximal allocatable resource. Also hard to understand why need to traverse map. Why not do something like this:

1. If sum of share pool sizes do not exceed non binding numa resources, do nothing.
2. Else restrict share pools in proportion as line 550 does, then return.

Reply:
1. Already refined it this way.
2. If CPU is assigned to the multi share regions in proportion, the sum of share pool sizes may not be exactly equal to their maximal allocatable resource, so we need to make some adjustments.

Reply:
If we have 569-571, it's more reasonable to remove line 568?